app/controllers/fuel/posts_controller.rb

```diff
@@ -3,13 +3,20 @@ class PostsController < FuelController
     include ActionView::Helpers::TextHelper
     layout Fuel.configuration.layout if Fuel.configuration.layout
     before_filter :define_title
+    if Fuel.configuration.header_pagination
+      paginated_action only: [:index]
+    end
 
     def define_title
       @blog_title = Fuel.configuration.blog_title
     end
 
     def index
-      @posts = Fuel::Post.recent_published_posts.page(params[:page])
+      if Fuel.configuration.header_pagination
+        @posts = Fuel::Post.recent_published_posts.page(@pagination_current_page).per(@pagination_per_page)
```

Comment:
We should be able to determine enough during the pre-action filter (see my comments on `fuel_controller.rb`) that we can get away with using only this single line to retrieve Posts for all invocations of the action.